File: DataStructures/LinkedList/RecursiveSearch.py
class Node:
    def __init__(self,key):
        self.key = key
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
    def push(self, new_key):
        new_node = Node(new_key)
        new_node.next = self.head
        self.head = new_node

    def search(self, node, x):
        if(node == None):
            return False
        if(node.key == x):
            return True
        return self.search(node.next, x)
        #when calling recursive functions include self.function

if __name__ == '__main__':
    llist =  LinkedList()
    llist.push(24)
    llist.push(29)
    llist.push(25)
    llist.push(21)
    llist.push(32)

    print("yes") if llist.search(llist.head, 23) else print("no")

File: DataStructures/LinkedList/RecursiveSearch.java
class Node 
{
    int key;
    Node next;
    Node(int k)
    {
        key = k;
        next = null;
    }
}

class LinkedList
{
    Node head = null;

    public void push(int new_key)
    {
        Node new_node = new Node(new_key);
        new_node.next = head;
        head = new_node;
    }

    public boolean search(Node node, int x)
    {
        if(node == null)
        {
            return false;
        }

        if(node.key == x)
        {
            return true;
        }

        return search(node.next, x);
    }


    public static void main(String[] args)
    {
        LinkedList llist = new LinkedList();

        llist.push(2);
        llist.push(21);
        llist.push(8);
        llist.push(7);
        llist.push(4);

        llist.search(4) ? System.out.print("Yes"): System.out.print("No");
    }
}